http.ts

```diff
@@ -0,0 +1,130 @@
+import {log} from "./util";
+
+import { pubInternal, sub } from "./dispatch";
+import { main as pb } from "./msg.pb";
+
+const servers = new Map<number, Server>();
+const dec = new TextDecoder("utf8");
+const enc = new TextEncoder();
+
+export function initHttp() {
+  sub("http", (payload: Uint8Array) => {
+      const msg = pb.Msg.decode(payload);
+      const id = msg.httpServerId;
+      const s = servers.get(id);
+      s.onMsg(msg);
+  });
+}
+
+export interface RequestOptions {
+    method?: string;
+    referrer?: string;
+    mode?: string;
+    credentials?: string;
+    redirect?: string;
+    integrity?: string;
+    cache?: string;
+}
+
+export class Request {
+    opts: RequestOptions;
+    readonly defaultMethod = "GET";
+    path: string;
+    method: string;
+    body: any | string;
+    constructor(url: string, opts?: RequestOptions) {
+      if(opts == null) {
```

Comment:
style: there should be a space after the "if". Running "yarn lint --fix" should fix this, I think.